js/editor-controller.js

```diff
@@ -0,0 +1,293 @@
+/**
+ * The Editor can be controlled by remote hosts that wrap the editor with
+ * an iframe or in some kind of webview.
+ */
+'use strict';
+
+/**
+ * @returns An object with the fs wrapper.
+ */
+var EditorController = function() {
+    var _returnObj = {};
+
+    var _editorActions = null;
+    var _msgEventListener = null;
+    var _controllerHost = null;
+
+    /** Constants used for iframe messaging */
+    var CONTROLLER_MESSAGING = Object.freeze({
+        // Embed editor host type
+        type: 'pyeditor',
+        // Embed editor messaging actions
+        actions: {
+            // Editor sends it to then receive a project to load
+            // direction: Bidirectional
+            // dataOut: Nothing
+            // dataIn: projects - An array with python code, editor loads [0]
+            workspacesync: 'workspacesync',
+            // Notifies controller 'workspacesync' was successful
+            // direction: Output
+            // dataOut: Nothing
+            workspaceloaded: 'workspaceloaded',
+            // Sends the editor code to the controller, configured to do this
+            // periodically
+            // direction: Output
+            // dataOut: project - A string with the editor code
+            workspacesave: 'workspacesave',
+            // Controller sends code to load into the editor
+            // direction: Input
+            // dataIn: project - A string with python code
+            importproject: 'importproject',
+            // Controller sends a hex file to load into the editor
+            // direction: Input
+            // dataIn: filename - String with the hex file name
+            //         filestring - String with the hex contents
+            loadhex: 'loadhex',
+            // Controller sends a python file to load into the editor
+            // direction: Input
+            // dataIn: filename - String with the python file name
+            //         filestring - String with the python code from the file
+            loadfile: 'loadfile',
+            // Editor sends a python file to the controller
+            // direction: Output
+            // dataOut: filename - String with the python file name
+            //          filestring - String with the python code from the file
+            savefile: 'savefile',
+            // Editor sends a hex file to the controller
+            // direction: Output
+            // dataOut: filename - String with the hex file name
+            //          filestring - String with the python code from the file
+            flashhex: 'flashhex',
```

Comment:
@martinwork As all these commands are very similar I've consolidates `filestring`, `hexstring`, and `data` to all use the same term `filestring`.